### ISSUE TITLE
Custom Db.conf parameters get overwritten during upgrade, and fo-postinstall fails

### ISSUE THREAD

**Issue by alpianon** [fossology/fossology#2683]
### Description

In case of an upgrade of an existing installation, the Db.conf file gets overwritten with the default one. If db name, host, user or password are not the default ones, this will cause `fo-postinstall` to fail and fossology to stop working because of a db connection error. In such case, the correct parameters need to be put again into Db.conf, and `fo-postinstall` needs to be re-run.

This is a regression: in previous versions using make instead of cmake, this was handled correctly in [install/Makefile](https://github.com/fossology/fossology/blob/76450a21575d2a9ad3e5873b2f758fe408caf7a5/install/Makefile#L63C1-L65C68):

```bash
	@if [ ! -f $(DESTDIR)$(CONFPATH)/Db.conf -o "$(OVERWRITE)" ]; then \
		echo "NOTE: using default version for $(DESTDIR)$(CONFPATH)/Db.conf"; \
	 	$(INSTALL) -m 640 defconf/Db.conf $(DESTDIR)$(CONFPATH)/Db.conf; \
```

This is not the case anymore in [install/CMakeLists.txt](https://github.com/fossology/fossology/blob/8286d8aa0e02d22a54bf1010d65f20f9787f4bbb/install/CMakeLists.txt#L40):

```
    ${CMAKE_CURRENT_BINARY_DIR}/gen/Db.conf
```

#### How to reproduce

Upgrade an existing Fossology installation using make, with custom Db.conf parameters, to any new version using cmake.

### Versions

It should affect all versions since 4.3.0 (CMake)

**Comment by GMishx** [comment]
I agree @alpianon . CMake should avoid overwriting conf files (extension `.conf` and `.in`). There should be a dedicated flag to force the overwrite, otherwise, the files should not be touched.

**Comment by dvjsharma** [comment]
I'll look into this

**Comment by dvjsharma** [comment]
Hi @GMishx, 

I've studied the CMake build system structure at FOSSology and in the existing installation process, when upgrading FOSSology, the `.conf` files in `usr/local/etc/fossology` are overwritten with the default version during the build process, leading to `fo-postinstall` failures and, consequently, FOSSology stopping due to database connection errors.

This can be solved by following the steps below:
* **Introducing `FORCE_CONF_OVERWRITE` Option** : This option will be introduced in the `CMakeLists.txt` file located in the `/install` sub-directory. Additionally, it will be declared in the top-level `CMakeLists.txt` file from where it can be propagated to `/install` sub-directory. 

* **Checking based on option value**:
Refer the snippet below
https://github.com/fossology/fossology/blob/8286d8aa0e02d22a54bf1010d65f20f9787f4bbb/install/CMakeLists.txt#L38-L43
This will be replaced by 
```cmake
if (NOT EXISTS "${FO_SYSCONFDIR}/Db.conf" OR FORCE_CONF_OVERWRITE)
    message(STATUS "Installing Db.conf")
    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/gen/Db.conf
            COMPONENT common
            DESTINATION ${FO_SYSCONFDIR})
else()
    message(STATUS "Db.conf already exists. Not overwriting, consider checking it by hand or use the FORCE_CONF_OVERWRITE option.")
endif()

if (NOT EXISTS "${FO_SYSCONFDIR}/fossology.conf" OR FORCE_CONF_OVERWRITE)
    message(STATUS "Installing fossology.conf")
    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/gen/fossology.conf
            COMPONENT common
            DESTINATION ${FO_SYSCONFDIR})
else()
    message(STATUS "fossology.conf already exists. Not overwriting, consider checking it by hand or use the FORCE_CONF_OVERWRITE option.")
endif()
```
This will ensure if `FORCE_CONF_OVERWRITE` option is not specified, the conf files are not touched.

I've tested the above solution on my local by changing the db password and running the scripts. They work fine. If you find the solution correct I'll open a PR.

Additionally, I wanted your advice on should we exclude overwriting more files as we did earlier in [Makefile](https://github.com/fossology/fossology/blob/76450a21575d2a9ad3e5873b2f758fe408caf7a5/install/Makefile )? PTAL
Thanks!

**Comment by GMishx** [comment]
Solutions sounds interesting @dvjsharma . Please check if you can write a function out of it, as the if-else are increasing lines of code by a lot.

Also, yes, we should consider more flies like Db.conf, fossology.conf, other email header and footer files, etc.

/cc @avinal
